src/training/ea/lipizzaner_trainer.py

```diff
@@ -69,40 +71,112 @@ def tournament_selection(
+            self.fitness_iterator = iter(
+                self.fitness_loaded
+            )  # Create iterator for fitness loader
+
+            # Determine how to aggregate fitness calculated among neighbourhood
+            self.fitness_mode = trainer_params["fitness"].get(
+                "fitness_mode", fitness_mode
             )
-        # ADAM Learning Rate configuration
+            if self.fitness_mode not in ["worst", "best", "average"]:
+                raise NotImplementedError(
+                    "Invalid argument for fitness_mode: {}".format(self.fitness_mode)
+                )
+        else:
+            # TODO: Add code for safe implementation & error handling
+            raise KeyError("Fitness section must be defined in configuration file")
+
+        if (
+            "score" in trainer_params
+            and trainer_params["score"].get("enabled", calc_mixture)
+        ) or "optimize_mixture" in trainer_params:
+            self.score_calc = ScoreCalculatorFactory.create()
+            self.score_sample_size = trainer_params["score"].get(
+                "sample_size", score_sample_size
+            )
+            self.score = float("inf") if self.score_calc.is_reversed else float("-inf")
+            self.mixture_generator_samples_mode = self.cc.settings["trainer"][
+                "mixture_generator_samples_mode"
+            ]
+        else:
+            self.score_sample_size = score_sample_size
+            self.score_calc = None
+            self.score = 0
 
-        self._default_adam_learning_rate = self.settings.get(
-            "default_adam_learning_rate", default_adam_learning_rate
-        )
+        if "optimize_mixture" in trainer_params:
+            self.optimize_weights_at_the_end = trainer_params["optimize_mixture"].get(
+                "enabled", True
+            )
+            self.score_sample_size = trainer_params["optimize_mixture"].get(
+                "sample_size", es_score_sample_size
+            )
+            self.es_generations = trainer_params["optimize_mixture"].get(
+                "es_generations", es_generations
+            )
+            self.es_random_init = trainer_params["optimize_mixture"].get(
+                "es_random_init", es_random_init
+            )
+            self.mixture_sigma = trainer_params["optimize_mixture"].get(
+                "mixture_sigma", mixture_sigma
+            )
+            self.mixture_generator_samples_mode = self.cc.settings["trainer"][
+                "mixture_generator_samples_mode"
+            ]
+        else:
+            self.optimize_weights_at_the_end = True
+            self.score_sample_size = es_score_sample_size
+            self.es_generations = es_generations
+            self.es_random_init = es_random_init
+            self.mixture_sigma = mixture_sigma
+            self.mixture_generator_samples_mode = self.cc.settings["trainer"][
+                "mixture_generator_samples_mode"
+            ]
```

Comment:
What about moving this to a separate helper `_configure_mixture(...)` or something like that?

Initialization methods all over Lipi are overencumbered by the way they implemented the configuration accessor, would be nice to avoid that happening on our code.

Same for other similar chunks